services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/S3ResumeToken.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.multipart;
+
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@SdkInternalApi
+public class S3ResumeToken {
+
+    private final String uploadId;
+    private final long partSize;
+    private final long totalNumParts;
+    private final long numPartsCompleted;
+
+    public S3ResumeToken(String uploadId, long partSize, long totalNumParts, long numPartsCompleted) {
```

Comment:
nit: can we use builder pattern? It's error prone when we have so many parameters with the same type.